.github/actions/check_docstrings/action.yml

```diff
@@ -0,0 +1,63 @@
+name: Check Docstrings
+inputs:
+    python-version:
+        description: 'The version of Python to use for the workflow.'
+        default: '3.10'
+        required: false
+        type: string
+    repository:
+        description: 'The repository to check the docstrings for.'
+        required: True
+        type: string
+    branch:
+        description: 'The branch to check the docstrings for.'
+        required: false
+        type: string
+        default: ''
+    package-name:
+        description: 'The name of the package to check the docstrings for.'
+        required: True
+        type: string
+    names-to-skip:
+        description: 'The names of the modules/classes/functions/methods to skip in the docstring check.'
+        required: false
+        type: string
+        default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Conda
+      uses: s-weigand/setup-conda@v1
+
+    - name: Setup Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Global Setup
+      run: |
+        pip install -U pip
+        pip install pytest-xdist
+        git config --global user.email "CI@example.com"
+        git config --global user.name "CI Almighty"
+      shell: bash
+
+    - name: Checkout Repository to be tested
+      uses: actions/checkout@v2
+      with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.branch }}
+
+    - name: Install package
+      run: pip install .
+      shell: bash
+
+    - name: Checkout Home Repository
```

Comment:
I think that if you put the files in the action folder you don't need to use the checkout actions to get the full repository.

Reply:
No, I think I still need the checkout actions -- see https://github.com/catalystneuro/.github/actions/runs/10565870849/job/29271332750